tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -578,10 +578,12 @@ ts_decompress_chunk_generate_paths(PlannerInfo *root, RelOptInfo *chunk_rel, Hyp
 
 	/*
 	 * since we rely on parallel coordination from the scan below
-	 * this node it is probably not beneficial to have more
-	 * than a single worker per chunk
+	 * request the same amount of parallel workers here as
+	 * PostgreSQL uses for the parallel scan on the compressed chunk
 	 */
-	int parallel_workers = 1;
+	int parallel_workers =
+		compute_parallel_worker(chunk_rel, chunk_rel->pages, -1, max_parallel_workers_per_gather);
```

Comment:
Can we have an index scan here? Might make sense to pass a proper value for index pages, and add a test for it.

Reply:
Regarding index scan in Timescale we are rely on PostgreSQL.
So no changes had been made. 

We use the function set_plain_rel_pathlist() 
which calls PostgreSQL function create_index_paths()
https://github.com/timescale/timescaledb/blob/main/src/import/allpaths.c#L129-L130

create_index_paths() calls get_index_paths() which calls build_index_paths()
Finally build_index_paths() function constructs zero or more partial IndexPaths

https://github.com/postgres/postgres/blob/ccd3623256220b944d9da00df75d91ef4d550362/src/backend/optimizer/path/indxpath.c#L235
https://github.com/postgres/postgres/blob/ccd3623256220b944d9da00df75d91ef4d550362/src/backend/optimizer/path/indxpath.c#L733
https://github.com/postgres/postgres/blob/ccd3623256220b944d9da00df75d91ef4d550362/src/backend/optimizer/path/indxpath.c#L855
https://github.com/postgres/postgres/blob/ccd3623256220b944d9da00df75d91ef4d550362/src/backend/optimizer/path/indxpath.c#L816

Reply:
Regarding the parallel index scan test case a test case has been added, 
which shows that PostgreSQL allocates 3 Workers for a Parallel Index Scan.

```
QUERY PLAN                                                                                                                                                                           
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Gather
   Output: _hyper_37_71_chunk."time", _hyper_37_71_chunk.sensor_id, _hyper_37_71_chunk.cpu, _hyper_37_71_chunk.temperature
   Workers Planned: 3
   ->  Parallel Append
         ->  Custom Scan (DecompressChunk) on _timescaledb_internal._hyper_37_71_chunk
               Output: _hyper_37_71_chunk."time", _hyper_37_71_chunk.sensor_id, _hyper_37_71_chunk.cpu, _hyper_37_71_chunk.temperature
               ->  Parallel Index Scan using compress_hyper_38_72_chunk__compressed_hypertable_38_sensor_id_ on _timescaledb_internal.compress_hyper_38_72_chunk
                     Output: compress_hyper_38_72_chunk."time", compress_hyper_38_72_chunk.sensor_id, compress_hyper_38_72_chunk.cpu, compress_hyper_38_72_chunk.temperature, compress_hyper_38_72_chunk._ts_meta_count, compress_hyper_38_72_chunk._ts_meta_sequence_num, compress_hyper_38_72_chunk._ts_meta_min_1, compress_hyper_38_72_chunk._ts_meta_max_1
                     Index Cond: (compress_hyper_38_72_chunk.sensor_id > 2)
(9 rows)
```